cmd/router/main.go

```diff
@@ -27,11 +32,39 @@ func callService(serviceUrl string, input []byte, res chan<- []byte) error {
 	return nil
 }
 
+func pickupRoute(routes []v1alpha1.InferenceRoute) *v1alpha1.InferenceRoute {
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+	//generate num [0,100)
+	point := r.Intn(99)
+	start, end := 0, 0
+
+	for _, route := range routes {
+		start += end
+		end += int(*route.Weight)
+		if point >= start && point < end {
+			return &route
+		}
+	}
+	return nil
+}
+
 func routeStep(nodeName string, currentStep v1alpha1.InferenceRouter, graph v1alpha1.InferenceGraphSpec, input []byte, res chan<- []byte) error {
 	log.Printf("current step %v", nodeName)
 	response := map[string]interface{}{}
 	//For splitter and ABNTest call virtual service
 	if currentStep.RouterType == v1alpha1.Splitter {
+		route := pickupRoute(currentStep.Routes)
+		if route == nil {
```

Comment:
in which case it could be nil ?

Reply:
as the logic, the `route` will never be `nil`, let me remove this check.